kite-data/kite-data-core/src/main/java/org/kitesdk/data/DatasetDescriptor.java

```diff
@@ -982,7 +981,7 @@ private static void checkPartitionStrategy(
         }
 
         // check the entity is a record if there is a non-provided partitioner
-        Preconditions.checkState(schema.getType() == Schema.Type.RECORD,
+        ValidationException.check(schema.getType() == Schema.Type.RECORD,
```

Comment:
While this was already a problem, should we switch to passing objects rather than building the string here? This serializes the schema every time the function runs.